### PR TITLE
nfdump support for with NFSen

### PR DESCRIPTION
**Pull request by VVelox**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

For those of you who are curious as to what this looks like.

image

Was originally https://github.com/librenms/librenms/pull/9987 , but that was a old one from when html moved to include/html and the merge did not totally go nicely.

### CODE REVIEW

File: includes/html/functions.inc.php [hunk 1]
```diff
@@ -1678,3 +1678,154 @@ function get_unit_for_sensor_class($class)
...
+
+/**
+ * @params string hostname
+ * @return array
+ *
+ * Returns a list of channels/rrds for a specified hostname. The
+ * keys for the string are channel names and the values are is the
+ * path to the RRD for the channel.
+*/
+function nfsen_channel_rrds($hostname)
```
Comment (murrant) [inline includes/html/functions.inc.php]: This function is not used.

File: includes/html/functions.inc.php [hunk 2]
```diff
@@ -1678,3 +1678,154 @@ function get_unit_for_sensor_class($class)
 
     return $units_by_classes[$class];
 }
+
+/**
+ * Rounds down to the closest number divisible by 5.
+ */
+function lowest_five($number)
```
Comment (murrant) [inline includes/html/functions.inc.php]: This function is not used.

File: includes/html/functions.inc.php [hunk 3]
```diff
@@ -1678,3 +1678,154 @@ function get_unit_for_sensor_class($class)
...
+{
+    return floor($number / 5) * 5;
+}
+
+/**
+ * Rounds down to the nearest 5 minutes if using seconds.
+ */
+function lowest_five_minutes($time)
```
Comment (murrant) [inline includes/html/functions.inc.php]: This function could be made more generic adding the interval as a second optional argument defaulting to 300

File: includes/html/pages/device/nfsen/stats.inc.php [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+print_optionbar_start();
+
+echo '<form action="'.generate_url($link_array, array('nfsen' => 'stats')).'" id="FlowStats" method="SUBMIT">';
+
+echo 'Top N:
+<select name="topN" id="topN" size=1>
+';
+
+$option_default=\LibreNMS\Config::get('nfsen_top_default');
```
Comment (murrant) [inline includes/html/pages/device/nfsen/stats.inc.php]: These three lines could be: 
```
$option_default = $vars['topN'] ?? \LibreNMS\Config::get('nfsen_top_default');
```

File: includes/html/graphs/device/nfsen_common.inc.php [hunk 5]
```diff
@@ -2,20 +2,18 @@
 
 $simple_rrd = true;
 
-foreach ((array)\LibreNMS\Config::get('nfsen_rrds', []) as $nfsenrrds) {
+
+$nfsen_rrds_array=\LibreNMS\Config::get('nfsen_rrds');
```
Comment (murrant) [inline includes/html/graphs/device/nfsen_common.inc.php]: Why did you change this?  The one on the left does the same thing.

File: includes/html/graphs/device/nfsen_channel_common.inc.php [hunk 6]
```diff
@@ -2,38 +2,48 @@
...
 $simple_rrd = true;
 
-foreach ((array)\LibreNMS\Config::get('nfsen_rrds', []) as $nfsenrrds) {
-    $nfsenrrds = rtrim($nfsenrrds, '/') . '/';
 
-    if (\LibreNMS\Config::get('nfsen_split_char')) {
-        $nfsenHostname = str_replace('.', \LibreNMS\Config::get('nfsen_split_char'), $device['hostname']);
-    } else {
-        $nfsenHostname=$device['hostname'];
+$nfsen_rrds_array=\LibreNMS\Config::get('nfsen_rrds');
```
Comment (murrant) [inline includes/html/graphs/device/nfsen_channel_common.inc.php]: Again, previous code was simpler.

**Comment by VVelox** [comment]
@murrant Howdy! All done with that. Sorry about the delay.

File: includes/html/functions.inc.php [hunk 7]
```diff
@@ -34,7 +34,7 @@ function var_eq($v, $t)
 /**
  * Get the value of $vars[$v], if it exists
  * @param string $v Name of the var to get
- * @return string|boolean The value of $vars[$v] if it exists, false if it does not exist
+ * @return string|boolean The value of $vars[$v] if it exists, false if it does not existhttps://static1.e621.net/data/23/6f/236f42f64c81fc264f927e64978ef7f6.jpg
```
Comment (laf) [inline includes/html/functions.inc.php]: Think that url might have landed in by accident.

**Comment by laf** [comment]
Sounds petty but could you update the formatting where you've done `something=nothing` to `something = nothing`. Also for the html code `OPTION` to lower case `option`.

File: includes/html/functions.inc.php (hunk 7 above)
Comment (VVelox) [reply includes/html/functions.inc.php]: Derp. =^.^=

Yeah. Removed. :)

**Comment by VVelox** [comment]
@laf Sure. Null sweat. :)

I'll get it knocked out tonight.

**Comment by VVelox** [comment]
@laf All done. :)

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-55-release-changelog-august-2019/9428/1
